lms/djangoapps/course_home_api/outline/v1/views.py

```diff
@@ -53,8 +60,23 @@ def get(self, request, course_key_string):
         course_key = CourseKey.from_string(course_key_string)
         course_tools = CourseToolsPluginManager.get_enabled_course_tools(request, course_key)
 
+        course = get_course_with_access(request.user, 'load', course_key, check_if_enrolled=False)
+        blocks = get_course_date_blocks(course, request.user, request, num_assignments=1)
+
+        # User locale settings
+        user_timezone_locale = user_timezone_locale_prefs(request)
+        user_timezone = user_timezone_locale['user_timezone']
+
+        dates_tab_link = reverse('dates', args=[course.id])
+        if course_home_mfe_dates_tab_is_active(course.id):
+            return get_microfrontend_url(course_key=course.id, view_name='dates')
```

Comment:
ee! We don't want to just return the microfrontend url because then we won't have any of the rest of the data.

So really we want this to be the value of `dates_tab_link`. You actually have two options here. You can either set the variable here and pass it in, or you can actually move this logic into the serializer and by using a SerializerMethodField to return the appropriate link there. An example of that format would be something like what happens in the CourseToolSerializer (plus some additional changes). But definitely your decision here. It is fantastic the way it is here, just trying to also let this be a learning opportunity. :)

Reply:
I think i may leave the link construction logic in here for readability/since it's only 2 lines of real code!